base/multi.jl

```diff
@@ -174,42 +164,22 @@ function flush_gc_msgs(w::Worker)
     end
 end
 
-# Acquire lock for write, recording the current task as owning the lock.
-# Returns true if lock was set, false if lock was already acquired
-function acquire_write_lock(w::Worker)
-    t = current_task()
-    while true
-        if isnull(w.w_io.locked_by)
-            w.w_io.locked_by = t; return true
-        else
-            if get(w.w_io.locked_by) == t
-                return false
-            end
-            wait(w.w_io.acq_lock_cond)
-        end
-    end
-end
-
-function release_write_lock(w::Worker)
-    w.w_io.locked_by = Nullable{Task}()
-    notify(w.w_io.acq_lock_cond)
-end
-
 function send_msg_(w::Worker, kind, args, now::Bool)
     #println("Sending msg $kind")
-    release_lock = acquire_write_lock(w)
+    io = w.w_stream
+    release_lock = lock(io)
```

Comment:
i feel like `lock` should be a `do-block` function, to ensure that it must get unlocked before any possible return

e.g
```julia
lock(io) do
    stuff
end
```